### PR TITLE
i18n(es): add configuration reference

### PR DESCRIPTION
**Pull request by kevinzunigacuellar**
#### What kind of changes does this PR include?

- Translated content

#### Description

- Closes add `reference/configuration-reference.md`
- add page to `nav.ts`

### CODE REVIEW

**Comment by Yan-Thomas** [comment]
Hey @kevinzunigacuellar, sorry for the delay. I'm personally going to review and merge your PR ASAP. Could you please update it to the most recent changes and fix the merge conflicts? Thanks for your amazing work! ❤️ https://github.com/withastro/docs/commits/main/src/pages/en/reference/configuration-reference.md

**Comment by kevinzunigacuellar** [comment]
> Hey @kevinzunigacuellar, sorry for the delay. I'm personally going to review and merge your PR ASAP. Could you please update it to the most recent changes and fix the merge conflicts? Thanks for your amazing work! ❤️ https://github.com/withastro/docs/commits/main/src/pages/en/reference/configuration-reference.md

I will get to it tonight! ❤️

File: src/pages/es/reference/configuration-reference.md [hunk 1]
```diff
@@ -0,0 +1,442 @@
...
+- 'server': construye una aplicación que se implementará en un host compatible con SSR (renderizado en el servidor).
+
+```js
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  output: 'static'
+})
+```
+**See Also:**
```
Comment (Yan-Thomas) [inline src/pages/es/reference/configuration-reference.md:200]: A small missing translation:
```suggestion
**Vea también:**
```

**Comment by Yan-Thomas** [comment]
LGTM, thanks for this @kevinzunigacuellar!

**Pull request merged**